WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipment Tracking Section/Add Tracking/ManualTrackingViewController.swift

```diff
@@ -230,58 +230,66 @@ extension ManualTrackingViewController: UITableViewDataSource {
     }
 
     private func configureShippingProvider(cell: TitleAndEditableValueTableViewCell) {
-        cell.title.text = NSLocalizedString("Shipping carrier", comment: "Add / Edit shipping carrier. Title of cell presenting name")
-        cell.value.text = viewModel.providerCellName
-        cell.value.placeholder = NSLocalizedString("Select carrier", comment: "Add the shipping carrier. Placeholder of cell presenting carrier name")
-
-        cell.value.isEnabled = false
+        let cellViewModel = TitleAndEditableValueTableViewCellViewModel(
+            title: NSLocalizedString("Shipping carrier", comment: "Add / Edit shipping carrier. Title of cell presenting name"),
+            placeholder: NSLocalizedString("Select carrier", comment: "Add the shipping carrier. Placeholder of cell presenting carrier name"),
+            initialValue: viewModel.providerCellName,
+            allowsEditing: false
+        )
+        cell.update(viewModel: cellViewModel)
         cell.accessoryType = viewModel.providerCellAccessoryType
     }
 
     private func configureProviderName(cell: TitleAndEditableValueTableViewCell) {
-        cell.title.text = NSLocalizedString("Carrier name", comment: "Add Custom shipping carrier. Title of cell presenting the carrier name")
-        cell.value.placeholder = NSLocalizedString("Enter carrier name", comment: "Add custom shipping carrier. Placeholder of cell presenting carrier name")
-
-        cell.value.text = viewModel.providerName
-        cell.value.isEnabled = true
-
-        cell.value.addTarget(self, action: #selector(didChangeProviderName), for: .editingChanged)
+        let cellViewModel = TitleAndEditableValueTableViewCellViewModel(
+            title: NSLocalizedString("Carrier name", comment: "Add Custom shipping carrier. Title of cell presenting the carrier name"),
+            placeholder: NSLocalizedString("Enter carrier name", comment: "Add custom shipping carrier. Placeholder of cell presenting carrier name"),
+            initialValue: viewModel.providerName
+        )
+        cell.update(viewModel: cellViewModel)
         cell.accessoryType = .none
+
+        _ = cellViewModel.value.subscribe { [weak self] in
```

Comment:
Don't these subscriptions have to be stored somewhere? 🤔

Reply:
I could have made a mistake but I looked at the Memory Graph after this VC is dismissed and these objects are all deallocated. So I believe it's fine. 

The block is also not retaining `self` so it's only owned by the `Observable` (`BehaviorSubject`).